File: archive/logic/TuringMachineSimulationNew/States/Transition/TransitionStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TuringMachineSimulation.Logic.States.Transition
{
    /// <summary> The transition one head should perform. </summary>
    public class HeadTransition
    {
        public char Read;
        public char Write;
        public int Moves;

        public HeadTransition(char read, char write, int moves)
        {
            Read = read;
            Write = write;
            Moves = moves;
        }
    }
    
    /// <summary>
    /// TransitionStatement represents one possible transitions for all heads.
    /// This data structure is not used for simulation, but a simpler representation for UI.
    /// </summary>
    public class TransitionStatement
    {
        public TransitionNode Source;
        public TransitionNode Target;

        public ICollection<HeadTransition> Conditions;

        public TransitionStatement(TransitionNode source, TransitionNode target, ICollection<HeadTransition> conditions)
        {
            Source = source;
            Target = target;
            Conditions = conditions;
        }
    }
}

File: frontend/turing-machine-web/src/logic/States/Transitions/TransitionStatement.ts
import { TransitionNode } from "./TransitionNode"

/** The transition one head should perform. */
export class HeadTransition
{
    public readonly Read: string = "";
    public readonly Write: string = "";
    public readonly Move: number = 0;

    constructor(read: string, write: string, move: number)
    {
        this.Read = read;
        this.Write = write;
        this.Move = move;
    }
}

/** 
 * TransitionStatement represents one possible transitions for all heads.
 * This data structure is not used for simulation, but only a simpler representation for UI.
 */
export class TransitionStatement
{
    public readonly Source: TransitionNode;
    public readonly Target: TransitionNode;
    public readonly Conditions: HeadTransition[];

    /** Constructor to create a new statement. All built by UI. */
    constructor(source: TransitionNode, target: TransitionNode, conditions: HeadTransition[])
    {
        this.Source = source;
        this.Target = target;
        this.Conditions = conditions;
    }
}